File: bigdata/src/main/java/com/hl/bigdata/flink/stream/java/source/MySourceTime.java
package com.hl.bigdata.flink.stream.java.source;

import org.apache.commons.lang3.RandomUtils;
import org.apache.flink.streaming.api.functions.source.RichSourceFunction;

/**
 * @author huanglin
 * @date 2024/07/04 15:37
 */
public class MySourceTime extends RichSourceFunction<String>{
    @Override
    public void run(SourceContext<String> sourceContext) throws Exception {
        while(true) {
            String str = "hello" + RandomUtils.nextInt(1, 5);
            sourceContext.collect(str + "," + (System.currentTimeMillis() - RandomUtils.nextLong(100, 1000)));
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {

    }
}

File: bigdata/src/main/java/com/hl/bigdata/flink/stream/scala/source/MySourceTime.scala
package com.hl.bigdata.flink.stream.scala.source

import org.apache.commons.lang3.RandomUtils
import org.apache.flink.streaming.api.functions.source.{RichSourceFunction, SourceFunction}

/**
 * @author huanglin
 * @date 2024/07/05 11:04
 */
class MySourceTime extends RichSourceFunction[String]{

  override def run(sourceContext: SourceFunction.SourceContext[String]): Unit = {
    while(true) {
      val key = "hello" + RandomUtils.nextInt(1, 5);
      sourceContext.collect(key + "," + (System.currentTimeMillis() - RandomUtils.nextLong(1000, 30000)));
      Thread.sleep(1000);
    }
  }

  override def cancel(): Unit = ???
}
